gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-http-proxy/src/main/java/io/gravitee/plugin/endpoint/http/proxy/HttpProxyEndpointConnector.java

```diff
@@ -114,13 +130,18 @@ private boolean isGrpc(final HttpRequest httpRequest) {
         }
     }
 
-    @Override
-    protected void doStop() {
-        if (httpClientFactory != null) {
-            httpClientFactory.close();
-        }
-        if (grpcHttpClientFactory != null) {
-            grpcHttpClientFactory.close();
+    private Completable handleException(Throwable throwable, HttpExecutionContext ctx) {
+        if (
+            throwable instanceof TimeoutException ||
+            throwable instanceof NoStackTraceTimeoutException ||
+            throwable instanceof ReadTimeoutException ||
+            throwable instanceof ConnectTimeoutException
+        ) {
+            log.warn("Endpoint timeout for request: {}", ctx.request().path(), throwable);
+            return ctx.interruptWith(new ExecutionFailure(HttpStatusCode.GATEWAY_TIMEOUT_504));
         }
+
+        log.error("connector error ", throwable);
+        return Completable.error(throwable);
```

Comment:
You can return something similar to what InvokerAdapter does (for v2 with V4 emulation)

```java
return ctx.interruptWith(new ExecutionFailure(HttpStatusCode.BAD_GATEWAY_502).key(GATEWAY_CLIENT_CONNECTION_ERROR));
```